AppCenter/AppCenter/MSCustomProperties.m

```diff
@@ -92,4 +96,8 @@ - (BOOL)isValidValue:(NSObject *)value {
   return YES;
 }
 
+- (NSDictionary<NSString *, NSObject *> *)propertiesImmutableCopy {
```

Comment:
I think the allocation must be inside of a `@synchronized` block or there could be a race condition.

Reply:
Also, could you just call it `immutableCopy`?